src/main/java/io/confluent/connect/jdbc/source/JdbcSourceTask.java

```diff
@@ -334,8 +338,18 @@ protected void closeResources() {
         }
 
         if (results.isEmpty()) {
+          consecutiveEmptyResults++;
           log.trace("No updates for {}", querier.toString());
-          continue;
+
+          if (consecutiveEmptyResults >= CONSECUTIVE_EMPTY_RESULTS_BEFORE_RETURN) {
```

Comment:
Once it passes this check, it will start to tight loop with the framework. Perhaps we can set the context.timeout() to back-off slightly in this case.

Reply:
Also, rather than waiting for 3 empty results, we could wait for one pass through all of the table queries, so that tables with no updates don't prevent tables with updates from being serviced.

Reply:
I don't think we get a `timeout` in the source context to play with. The intent of the check is to try a bit to prevent going into the tight loop in framework. The idea to wait for a pass through all table queries seems good. I think I'll go a step further and require that all table queries have returned empty for at least the number of consecutive queries before returning control to framework.